packages/@aws-cdk/aws-glue/lib/security-configuration.ts

```diff
@@ -0,0 +1,239 @@
+
+  /**
+   * The encryption configuration for Amazon CloudWatch Logs.
+   * @default no cloudwatch logs encryption.
+   */
+  readonly cloudWatchEncryption?: CloudWatchEncryption,
+
+  /**
+   * The encryption configuration for Glue Job Bookmarks.
+   * @default no job bookmarks encryption.
+   */
+  readonly jobBookmarksEncryption?: JobBookmarksEncryption,
+
+  /**
+   * The encryption configuration for Amazon Simple Storage Service (Amazon S3) data.
+   * @default no s3 encryption.
+   */
+  readonly s3Encryption?: S3Encryption,
+}
+
+/**
+ * A security configuration is a set of security properties that can be used by AWS Glue to encrypt data at rest.
+ *
+ * The following scenarios show some of the ways that you can use a security configuration.
+ * - Attach a security configuration to an AWS Glue crawler to write encrypted Amazon CloudWatch Logs.
+ * - Attach a security configuration to an extract, transform, and load (ETL) job to write encrypted Amazon Simple Storage Service (Amazon S3) targets and encrypted CloudWatch Logs.
+ * - Attach a security configuration to an ETL job to write its jobs bookmarks as encrypted Amazon S3 data.
+ * - Attach a security configuration to a development endpoint to write encrypted Amazon S3 targets.
+ */
+export class SecurityConfiguration extends cdk.Resource implements ISecurityConfiguration {
+
+  /**
+   * Creates a Connection construct that represents an external connection.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param attrs Import attributes.
+   */
+  public static fromSecurityConfigurationAttributes(scope: constructs.Construct, id: string,
+    attrs: SecurityConfigurationAttributes): ISecurityConfiguration {
+
+    class Import extends cdk.Resource implements ISecurityConfiguration {
+      public readonly securityConfigurationName = attrs.securityConfigurationName;
+    }
+    return new Import(scope, id);
+  }
+
+  private static checkKmsKeyCompatibleWithMode(encryption?: {mode: string, kmsKey?: kms.IKey}): void {
+    if (encryption && /KMS/.test(encryption.mode) && encryption.kmsKey == undefined) {
+      throw new Error(`${encryption.mode} requires providing a kms key`);
+    }
+  }
+
+  /**
+   * The name of the security configuration.
+   * @attribute
+   */
+  public readonly securityConfigurationName: string;
+
+  constructor(scope: constructs.Construct, id: string, props: SecurityConfigurationProps) {
+    super(scope, id, {
+      physicalName: props.securityConfigurationName,
+    });
+
+    if (props.s3Encryption == undefined && props.cloudWatchEncryption == undefined && props.jobBookmarksEncryption == undefined) {
+      throw new Error('One of cloudWatchEncryption, jobBookmarksEncryption or s3Encryption must be defined');
+    }
+    SecurityConfiguration.checkKmsKeyCompatibleWithMode(props.cloudWatchEncryption);
+    SecurityConfiguration.checkKmsKeyCompatibleWithMode(props.jobBookmarksEncryption);
```

Comment:
We won't need these if we decide to create a key for the user.

Reply:
True, but see my comments above about this topic of creating the key